run_model_ghcr/__init__.py

```diff
@@ -0,0 +1,139 @@
+"""API endpoint for running the NHP Model"""
+
+import json
+import logging
+import re
+import zlib
+from datetime import datetime
+
+import azure.functions as func
+from azure.core.exceptions import ResourceExistsError
+from azure.identity import DefaultAzureCredential
+from azure.mgmt.containerinstance import ContainerInstanceManagementClient
+from azure.mgmt.containerinstance.models import (
+    Container,
+    ContainerGroup,
+    ContainerGroupIdentity,
+    ContainerGroupSubnetId,
+    EnvironmentVariable,
+    OperatingSystemTypes,
+    ResourceRequests,
+    ResourceRequirements,
+)
+from azure.storage.blob import BlobServiceClient
+
+import config
+
+
+def main(req: func.HttpRequest) -> func.HttpResponse:
+    """main api endpoint"""
+
+    params = req.get_json()
+    if "id" in params:
+        params.pop("id")
+    params["create_datetime"] = f"{datetime.utcnow():%Y%m%d_%H%M%S}"
+
+    metadata = {
+        k: str(v)
+        for k, v in params.items()
+        if not isinstance(v, dict) and not isinstance(v, list)
+    }
+    params["app_version"] = metadata["app_version"] = req.params.get(

```

Comment:
This is a positive change - at the moment the params in results can show a different version to the version actually used to run the model